src/core/src/index/revindex/mod.rs

```diff
@@ -425,7 +425,7 @@ impl Datasets {
         }
     }
 
-    fn from_slice(slice: &[u8]) -> Option<Self> {
+    pub fn from_slice(slice: &[u8]) -> Option<Self> {
```

Comment:
Since this is going `pub`, maybe make it

```suggestion
    pub fn from_slice(slice: &[u8]) -> Result<Self> {
```

instead and report errors (instead of `unwrap` everywhere like it is now)